sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -414,6 +414,16 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         // these branches can be pruned away
         val (h, t) = branches.span(_._1 != TrueLiteral)
         CaseWhen( h :+ t.head, None)
+
+      case CaseWhen(branches, elseValue) if branches.length == 1 =>
+        // Using pattern matching like `CaseWhen((cond, branchValue) :: Nil, elseValue)` will not
+        // work since the implementation of `branches` can be `ArrayBuffer`. A full test is in
```

Comment:
Just curious, when is it`ArrayBuffer`?

Reply:
`sql("select case when a is null then 1 end col1 from t")` will create `branches` with `ArrayBuffer` implementation.

Reply:
We can change it to immutable `List` to avoid confusion.

Reply:
@ueshin thanks! The code is much more cleaner.